### PR TITLE
Add function for testing that type can be resolved

### PR DESCRIPTION
**Pull request by kant2002**

### CODE REVIEW

**Comment by kant2002** [comment]
In case when unresolved type somewhere deep inside the struct we can known what type trigger cascading failures. That’s how original logic works. 

Reporting is still awful but I do not see how can I make proper infrastructure for it.

File: Cesium.CodeGen/Contexts/TranslationUnitContext.cs [hunk 1]
```diff
@@ -75,30 +99,84 @@ internal IType ResolveType(IType type)
...
+                }
+            }
+
+            resolvedType = new StructType(members, structType.Identifier);
+            return isStructResolved;
         }
 
         if (type is FunctionType functionType)
         {
+            bool isFunctionResolved = true;
```
Comment (ForNeVeR) [inline Cesium.CodeGen/Contexts/TranslationUnitContext.cs:134]: Ok, so there are a lot of warnings on that the variable is unused. Could you remove it then?

**Review by ForNeVeR — CHANGES_REQUESTED** [review]
I apporve this in principle, but could you please add a couple of tests for this new logic?

**Comment by ForNeVeR** [comment]
@kant2002 do you still think we need this? Sadly, I can't remember what the original intent was.

**Pull request closed without merging**